network-api/networkapi/wagtailpages/pagemodels/products.py

```diff
@@ -1343,10 +1371,11 @@ def get_context(self, request, *args, **kwargs):
                 self.cutoff_date,
                 authenticated,
                 key,
-                ProductPage.objects.all()
+                ProductPage.objects.exclude(product_categories__in=self.excluded_categories.all().values_list('id'))
```

Comment:
@Pomax this PR is still WIP but can I get a hand for this line? I don't know how to properly exclude ProductPages that fall under any categories we have in `excluded_categories`. This line does nothing and I also tried it without `.values_list('id')` but got 
```
ValueError at /en/privacynotincluded/

Cannot use QuerySet for "ExcludedCategories": Use a QuerySet for "ProductPageCategory".
```

Reply:
Note that this is now a "through" relation, so instead of `product_categories`, you need to test using `product_categories__category`, and the same goes for the exclude categories:

```
        ...
        products = cache.get(key)

        exclude_ids = [excats.excluded_category.id for excats in self.excluded_categories.all()]

        if not kwargs.get('bypass_products', False) and products is None:
            products = get_product_subset(
                self.cutoff_date,
                authenticated,
                key,
                ProductPage.objects.exclude(product_categories__category__id__in=exclude_ids)
            )

        ...
```

Reply:
although if you rename that field from `excluded_category` to just `category`, that will change the above code to:
```
exclude_ids = [excats.category.id for excats in self.excluded_categories.all()]
```

Reply:
Ahhh I see the differences now. Thanks for the explanation.